src/commands/reject.rs

```diff
@@ -1,29 +1,38 @@
+use crate::commands::WholeStreamCommand;
 use crate::errors::ShellError;
 use crate::object::base::reject_fields;
 use crate::prelude::*;
 
-pub fn reject(args: CommandArgs, registry: &CommandRegistry) -> Result<OutputStream, ShellError> {
-    let args = args.evaluate_once(registry)?;
-    let len = args.len();
-    let span = args.name_span();
-    let (input, args) = args.parts();
-
-    if len == 0 {
-        return Err(ShellError::labeled_error(
-            "Reject requires fields",
-            "needs parameter",
-            span,
-        ));
+#[derive(Deserialize)]
+pub struct RejectArgs {
+    rest: Vec<Tagged<String>>,
+}
+
+pub struct Reject;
+
+impl WholeStreamCommand for Reject {
+    fn run(
+        &self,
+        args: CommandArgs,
+        registry: &CommandRegistry,
+    ) -> Result<OutputStream, ShellError> {
+        args.process(registry, reject)?.run()
     }
 
-    let fields: Result<Vec<String>, _> = args
-        .positional
-        .iter()
-        .flatten()
-        .map(|a| a.as_string())
-        .collect();
+    fn name(&self) -> &str {
+        "reject"
+    }
+
+    fn signature(&self) -> Signature {
+        Signature::build("reject").rest()
+    }
+}
 
-    let fields = fields?;
+fn reject(
+    RejectArgs { rest: fields }: RejectArgs,
+    RunnableContext { input, .. }: RunnableContext,
+) -> Result<OutputStream, ShellError> {
+    let fields: Vec<_> = fields.iter().map(|f| f.item.clone()).collect();
```

Comment:
This (and other similar calls) doesn't feel right, maybe `reject_fields` should be updated to take `Vec<Tagged<String>>`?

Reply:
Yeah, agreed. It shouldn't care if they're tagged or not